internal/agent/cmdchannel/runintegration/runintegration.go

```diff
@@ -54,6 +56,10 @@ func NewHandler(definitionQ chan<- integration.Definition, il integration.Instan
 			return
 		}
 
+		if cmdapi.IsAllowedToRunStopFromCmdAPI(args.IntegrationName) {
+			return ErrIntNotAllowed
+		}
```

Comment:
Make sense to move it to be the first we validate. Therefore we avoid to 'Unmarshal' if there is no need.

Reply:
only previous check is empty case, and I think that's correct, as it should provide its own error, otherwise this  `IsAllowedToRunStopFromCmdAPI` should return a 3rd case (not bool anymore) where it says empty case is not handled